### DISCUSSION TITLE
Separating out #10 into smaller PRs

### DISCUSSION THREAD

**Discussion by jamied157** [synthesized-io/insight#22] [General]
After some discussion, I think we've settled on a structure for the repo that should start us off well

![Untitled Diagram](https://user-images.githubusercontent.com/16978470/130592704-7959cb00-8ba2-4cb6-9154-1019804f1873.png)

I think we can then split up PR #10  into parts that work towards this structure, I can think of these steps

1. Convert the current repo to this structure (i.e: add in p-values for the tests and move to new file)
2. Port in distance metrics from PR
    -  Binomial Distance
    -  Kullback Leibler Divergence
    -  Jensen Shannon Divergence
    -  Hellinger Distance
3. Port in correlation metrics from PR
    -  R2 Mcfadden
    -  Distance NN Correlation
    -  Distance CN Correlation
4. Add in extra tests
    - permutation tests
    -  Kruskal Wallis
    - Any more?
5. Add in Confidence interval calculations
    - Bootstrap Confidence Intervals
    - Binomial Proportion CI
    - Any more?

Let me know if I've missed anything, otherwise, we can start to split this work up between us.

**Comment by akanksha1304** [comment]
I think we can add another PR for adding more tests, as there are not many tests for bootstrap and permutation methods. But apart from that, the rest can be handled in one PR which I am working on

**Comment by jamied157** [comment]
Okay so the final update on this is that Akanksha is still happy to put this together as a single piece of work but will try and split up the PRs if she sees an easy way to do so.
